lib/portlayer/exec/handle.go

```diff
@@ -181,17 +186,15 @@ func (h *Handle) Commit(ctx context.Context, sess *session.Session, waitTime *in
 	cfg := make(map[string]string)
 
 	// Set timestamps based on target state
-	if h.State != nil {
-		switch *h.State {
-		case StateRunning:
-			se := h.ExecConfig.Sessions[h.ExecConfig.ID]
-			se.StartTime = time.Now().UTC().Unix()
-			h.ExecConfig.Sessions[h.ExecConfig.ID] = se
-		case StateStopped:
-			se := h.ExecConfig.Sessions[h.ExecConfig.ID]
-			se.StopTime = time.Now().UTC().Unix()
-			h.ExecConfig.Sessions[h.ExecConfig.ID] = se
-		}
+	switch h.CurrentState() {
```

Comment:
Seeing this here I really think we should not be overloading the imperative state field in the handle, with the descriptive state field in the container. They have different valid values and are used for different things. This is going to lead to even more confusion in the future.